network/src/compress.rs

```diff
@@ -0,0 +1,178 @@
+use bytes::{Bytes, BytesMut};
+use snap::{Decoder as SnapDecoder, Encoder as SnapEncoder};
+use tokio::codec::{Decoder, Encoder, LengthDelimitedCodec};
+
+use std::io;
+
+const SKIP_COMPRESS_SIZE: usize = 40 * 1024;
+
+/// Compressed decompression structure
+///
+/// If you want to support multiple compression formats in the future,
+/// you can simply think that 0b1000 is in snappy format and 0b0000 is in uncompressed format.
+///
+/// # Message in Bytes:
+///
+/// +---------------------------------------------------------------+
+/// | Bytes | Type | Function                                       |
+/// |-------+------+------------------------------------------------|
+/// |   0   |  u1  | Compress: true 1, false 0                      |
+/// |       |  u7  | Reserved                                       |
+/// +-------+------+------------------------------------------------+
+/// |  1~   |      | Payload (Serialized Data with Compress)        |
+/// +-------+------+------------------------------------------------+
+#[derive(Clone, Debug)]
+struct Message {
+    inner: BytesMut,
+}
+
+impl Message {
+    /// Init
+    fn init() -> Self {
+        Message {
+            inner: BytesMut::from(&[0u8][..]),
+        }
+    }
+
+    /// Compress message
+    fn compress(&mut self, input: Bytes) {
```

Comment:
The way to create an empty message, then compress the input into the message looks strange to me. For example, we may forget to create new message, and call `compress` multiple times on the message. It's better to be a constructor.

Reply:
Therefore, the method exported by this module is `compress` and a `codec`, and other modules cannot use the `Message` struct.

If it is really necessary to change it is also feasible